Userland/Libraries/LibWeb/MimeSniff/Resource.cpp

```diff
@@ -446,11 +446,13 @@ ErrorOr<void> Resource::supplied_mime_type_detection_algorithm(StringView scheme
     // NOTE: Non-standard but this algorithm expects the caller to handle step 2.1.1.
     if (supplied_type.has_value()) {
         if (Fetch::Infrastructure::is_http_or_https_scheme(scheme)) {
+            // NOTE: The spec expects a space between the semicolon and the start of the charset parameter. However, we will lose this
+            //       space because MimeType::parse() ignores any spaces found there.
             static Array<StringView, 4> constexpr apache_bug_mime_types = {
                 "text/plain"sv,
-                "text/plain; charset=ISO-8859-1"sv,
-                "text/plain; charset=iso-8859-1"sv,
-                "text/plain; charset=UTF-8"sv
+                "text/plain;charset=ISO-8859-1"sv,
+                "text/plain;charset=iso-8859-1"sv,
+                "text/plain;charset=UTF-8"sv
```

Comment:
I wanted to point particular attention to this change as I had to remove the space between the semicolon and the `charset` parameter due to the MimeType parsing algorithm. However, the spec expects us to match **exactly** with the byte sequences they have provided, where the last 3 types provided include a space.

I'm not sure if I need to rethink the supplied type detection algorithm because of this, but from my point of view `text/plain; charset=UTF-8` and `text/plain;charset=UTF-8` represent the exact same MIME type. This is how the MimeType [parsing algorithm](https://mimesniff.spec.whatwg.org/#parse-a-mime-type) sees it too (read the only Note for more details). Just not sure if there is something I'm overlooking here 😅

Reply:
Sounds like a good case for a spec issue. If we can come up with a small test case that shows the difference between what the web engines do and the odd space in the spec that seems to violate the assumptions of another spec algorithm, then it might be a real bug? Otherwise, it's probably an editorial fix we could PR to the spec.

Reply:
I haven't found anything regarding how Firefox checks for the Apache bug, but did find [this](https://github.com/chromium/chromium/blob/89750fbbc060bcc87e9747b25c80db6e240ce2ec/net/base/mime_sniffer.cc#L751C36-L751C46) from Chrome's implementation. It appears to just check if the raw String is "text/plain" (so it doesn't even consider the charset parameter), and if so it runs the [rules for distinguishing if a resource is a text or binary](https://mimesniff.spec.whatwg.org/#sniffing-a-mislabeled-binary-resource).

[servo](https://github.com/servo/servo/blob/21d9c2cc637842118adc5af682422e3df994710b/components/net/mime_classifier.rs#L33C4-L33C4) on the other hand does seem to grab the raw result of `Content-Type` to check if it should set the Apache bug flag.

From this, and seeing that my tests use the exact byte sequences that the spec requires, I'm leaning more towards our implementation should be fine 🤔?

Reply:
sounds like a good Web Platform Test to write then :P (https://github.com/web-platform-tests/wpt) (or check if they have one for this already)